### PR TITLE
[Kernel][FusedMoE] Fix MoE crash and hang issues

### PR DESCRIPTION
**Pull request by bythew3i**
... also move weight padding to weight load instead of inside kernel.

# Root Cause:

1. TopK on padded gating scores could return out of range topK index
2. Missing sync_barrier to make sure all devices finish metadata propagation before a2a scatter.
3. Large `bt` (block num tokens) caused OOB read and write

# TODO:
- Tune the block sizes  in MoE for GPT-OSS

### CODE REVIEW

File: tpu_inference/runner/kv_cache.py
Comment (kyuyeunk) [inline tpu_inference/runner/kv_cache.py:1]: we need a different PR for this. I found out it's not simple replacement because this function doesn't exist in order jax versions and causes crash.

File: tpu_inference/layers/vllm/quantization/unquantized.py [hunk 1]
```diff
@@ -336,6 +377,11 @@ def apply(
         gating_output = jax_view(router_logits)
 
         if self.use_kernel and layer.use_ep:
```
Comment (kyuyeunk) [inline tpu_inference/layers/vllm/quantization/unquantized.py]: I think this branch is outdated - because i see old code. can you update again?

File: tpu_inference/runner/kv_cache.py
Comment (bythew3i) [reply tpu_inference/runner/kv_cache.py:1]: Yes, agree, looks like we need to bump jax version in tpu-inference. Google internal dev env is always using latest jax

File: tpu_inference/layers/vllm/quantization/unquantized.py (hunk 1 above)
Comment (bythew3i) [reply tpu_inference/layers/vllm/quantization/unquantized.py]: Synced to main. I do not see any difference or conflict here with main branch

File: tpu_inference/runner/kv_cache.py
Comment (bythew3i) [reply tpu_inference/runner/kv_cache.py:1]: Reverts the item_width change

File: tpu_inference/kernels/fused_moe/v1/kernel.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def align_to(x, a):
 
 
 def get_dtype_packing(dtype):
-    bits = dtypes.bit_width(dtype)
+    bits = dtypes.itemsize_bits(dtype)
```
Comment (kyuyeunk) [inline tpu_inference/kernels/fused_moe/v1/kernel.py]: This change should also be removed

**Comment by kyuyeunk** [comment]
@bythew3i Can you update the branch & resolve the conflict?
